tmt/steps/provision/__init__.py

```diff
@@ -1278,6 +1278,8 @@ def _ssh_options(self) -> Command:
                 options.extend(['-i', key])
         if self.password:
             options.extend(['-oPasswordAuthentication=yes'])
+        else:
+            options.extend(['-oPasswordAuthentication=no'])
```

Comment:
Perhaps it would be good to add at least a short explanation / justification here describing why this is actually needed, plus a link to the original issue?